### PR TITLE
DAFFODIL-2293

### PR DESCRIPTION
**Pull request by kuba72ch**
Two new encodings introduced: X-DFDL-ISO-88591-8-BIT-PACKED-LSB-FIRST, X-DFDL-ISO-88591-8-BIT-PACKED-MSB-FIRST. Additionally unit-tests und TDML-tests added.

### CODE REVIEW

File: daffodil-io/src/test/scala/org/apache/daffodil/io/TestNonByteSizedCharsetEncoders8Bit.scala [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+    val expectedBytes = Misc.bits2Bytes("00110000 00110001 00110010 00110011 00110100 00110101 00110110 00110111").toList
+    val bb = ByteBuffer.allocate(8)
+    val res = encoder.encode(cb, bb, false)
+    assertTrue(res.isUnderflow())
+    bb.flip()
+    val actualBytes = bb.array().toList
+    assertEquals(expectedBytes, actualBytes)
+  }
+
+  @Test def test3BitMSBF_02(): Unit = {
```
Comment (mbeckerle) [inline daffodil-io/src/test/scala/org/apache/daffodil/io/TestNonByteSizedCharsetEncoders8Bit.scala]: Change name to "8" Bit.

File: daffodil-io/src/test/scala/org/apache/daffodil/io/TestNonByteSizedCharsetEncoders8Bit.scala [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+
+import org.junit.Assert._
+import org.junit.Test
+
+import org.apache.daffodil.processors.charset.CharsetUtils
+import org.apache.daffodil.util.Misc
+
+class TestNonByteSizedCharsetEncoders8Bit {
+
+  @Test def test3BitMSBF_01(): Unit = {
```
Comment (mbeckerle) [inline daffodil-io/src/test/scala/org/apache/daffodil/io/TestNonByteSizedCharsetEncoders8Bit.scala]: Change name to 8Bit

File: daffodil-io/src/test/scala/org/apache/daffodil/io/TestNonByteSizedCharsetEncoders8Bit.scala [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+    val expectedBytes = Misc.bits2Bytes("00110000 00110001 00110010 00110011 00110100 00110101 00110110 00110111").toList
+    val bb = ByteBuffer.allocate(8) // not enough space for last digit
+    val res = encoder.encode(cb, bb, false)
+    assertTrue(res.isOverflow())
+    bb.flip()
+    val actualBytes = bb.array().toList
+    assertEquals(expectedBytes, actualBytes)
+  }
+
+  @Test def test3BitLSBF_01(): Unit = {
```
Comment (mbeckerle) [inline daffodil-io/src/test/scala/org/apache/daffodil/io/TestNonByteSizedCharsetEncoders8Bit.scala]: change name to 8 bit

File: daffodil-io/src/test/scala/org/apache/daffodil/io/TestNonByteSizedCharsetEncoders8Bit.scala [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+    val expectedBytes = Misc.bits2Bytes("00110000 00110001 00110010 00110011 00110100 00110101 00110110 00110111").toList
+    val bb = ByteBuffer.allocate(8)
+    val res = encoder.encode(cb, bb, false)
+    assertTrue(res.isUnderflow())
+    bb.flip()
+    val actualBytes = bb.array().toList
+    assertEquals(expectedBytes, actualBytes)
+  }
+
+  @Test def test3BitLSBF_02(): Unit = {
```
Comment (mbeckerle) [inline daffodil-io/src/test/scala/org/apache/daffodil/io/TestNonByteSizedCharsetEncoders8Bit.scala]: and one more. Name "8" bit

File: daffodil-io/src/test/scala/org/apache/daffodil/io/TestNonByteSizedCharsetEncoders8Bit.scala (hunk 1 above)
Comment (kuba72ch) [reply daffodil-io/src/test/scala/org/apache/daffodil/io/TestNonByteSizedCharsetEncoders8Bit.scala]: Unit-test-names renamed from 3- to 8-bit (for decoder and encoder).

File: daffodil-io/src/test/scala/org/apache/daffodil/io/TestNonByteSizedCharsetEncoders8Bit.scala (hunk 2 above)
Comment (kuba72ch) [reply daffodil-io/src/test/scala/org/apache/daffodil/io/TestNonByteSizedCharsetEncoders8Bit.scala]: Unit-test-names renamed from 3- to 8-bit (for decoder and encoder).

File: daffodil-io/src/test/scala/org/apache/daffodil/io/TestNonByteSizedCharsetEncoders8Bit.scala (hunk 3 above)
Comment (kuba72ch) [reply daffodil-io/src/test/scala/org/apache/daffodil/io/TestNonByteSizedCharsetEncoders8Bit.scala]: Unit-test-names renamed from 3- to 8-bit (for decoder and encoder).

File: daffodil-io/src/test/scala/org/apache/daffodil/io/TestNonByteSizedCharsetEncoders8Bit.scala (hunk 4 above)
Comment (kuba72ch) [reply daffodil-io/src/test/scala/org/apache/daffodil/io/TestNonByteSizedCharsetEncoders8Bit.scala]: Unit-test-names renamed from 3- to 8-bit (for decoder and encoder).

**Comment by mbeckerle** [comment]
Can you please squash these two commits together?

That's our standard workflow, after review, squash all the fixup commits together into one commit. push that, then we merge.

**Pull request merged**